sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -1051,169 +1204,21 @@ static void temp_controller_invoke_reboot(void)
-            component_name,
-            property_name,
-            &property_value,
-            version,
-            publish_message.payload,
-            &publish_message.out_payload))
-    {
-      IOT_SAMPLE_LOG_ERROR(
-          "Temperature Controller does not support writable property `%.*s`. All writeable "
-          "properties are on sub-components.",
-          az_span_size(property_name->slice),
-          az_span_ptr(property_name->slice));
-
-      // Build the root component error message with status.
-      temp_controller_build_error_reported_property_with_status(
-          component_name,
-          property_name->slice,
-          &property_value,
-          AZ_IOT_STATUS_NOT_FOUND,
-          version,
-          publish_message.payload,
-          &publish_message.out_payload);
-    }
-  }
-  else
-  {
-    IOT_SAMPLE_LOG("No components recognized to update a property.");
-    return;
-  }
-
-  // Send response. On a successfull property update above, out_payload will already be set.
-  publish_mqtt_message(
-      publish_message.topic, publish_message.out_payload, IOT_SAMPLE_MQTT_PUBLISH_QOS);
-  IOT_SAMPLE_LOG_SUCCESS("Client sent reported property with status message.");
-  IOT_SAMPLE_LOG_AZ_SPAN("Payload:", publish_message.out_payload);
-  IOT_SAMPLE_LOG(" "); // Formatting.
-
-  // Receive the response from the server.
-  receive_mqtt_message();
-}
-
-static az_result append_int32_callback(az_json_writer* jw, void* value)
+static az_result append_json_token(az_json_writer* jw, az_json_token* value)
 {
-  return az_json_writer_append_int32(jw, *(int32_t*)value);
-}
-
-static az_result append_json_token_callback(az_json_writer* jw, void* value)
-{
-  char const* const log = "Failed to append json token callback";
+  char const* const log = "Failed to append json token";
 
   az_json_token value_token = *(az_json_token*)value;
 
-  double value_as_double;
-  int32_t string_length;
-
   switch (value_token.kind)
   {
-    case AZ_JSON_TOKEN_NUMBER:
-      IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_token_get_double(&value_token, &value_as_double), log);
-      IOT_SAMPLE_EXIT_IF_AZ_FAILED(
-          az_json_writer_append_double(jw, value_as_double, DOUBLE_DECIMAL_PLACE_DIGITS), log);
-      break;
     case AZ_JSON_TOKEN_STRING:
-      IOT_SAMPLE_EXIT_IF_AZ_FAILED(
-          az_json_token_get_string(
-              &value_token,
-              command_property_scratch_buffer,
-              sizeof(command_property_scratch_buffer),
-              &string_length),
-          log);
-      IOT_SAMPLE_EXIT_IF_AZ_FAILED(
-          az_json_writer_append_string(
-              jw, az_span_create((uint8_t*)command_property_scratch_buffer, string_length)),
-          log);
+      IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_writer_append_string(jw, value->slice), log);
       break;
     default:
-      return AZ_ERROR_ITEM_NOT_FOUND;
+      IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_writer_append_json_text(jw, value->slice), log);
```

Comment:
If you want to keep rejecting "complex" JSON tokens, that is fine, as a limitation in samples.

```C
switch (value_token.kind)
{
  case AZ_JSON_TOKEN_STRING:
    IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_writer_append_string(jw, value->slice), log);
    break;
  case null, true, false, number:
    IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_writer_append_json_text(jw, value->slice), log);
    break;
  default:
    return some not implemented/supportederror; // JSON value is complex {}, or [], needs to be implemented.
}
```